src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1182,6 +1208,59 @@ export class StudyViewPageStore
                             this.molecularProfileMapByType,
                             selectedSamples
                         );
+                    } else if (chartMeta.dataType === 'Custom_Data') {
+                        const clinicalAttribute = chartMeta.clinicalAttribute;
```

Comment:
lets move all this logic into a tested helper function that accepts chartMeta and sampleKeyToData as args

Reply:
This has been moved to the function `transformSampleDataToSelectedSampleClinicalData` in StudyViewutils.tsx  with a test in StudyViewUtils.spec.tsx.